pkg/kube/client.go

```diff
@@ -628,3 +641,89 @@ func (c *Client) watchPodUntilComplete(timeout time.Duration, info *resource.Inf
 
 	return err
 }
+
+//get the relation pods of kubernetes resources
+// kubernetes resource used select labels to relate pods
+func (c *Client) getRelationPods(infos []*resource.Info) map[string][]api.Pod {
+	var objPods = make(map[string][]api.Pod)
+
+	for _, value := range infos {
+		objPods, _ = c.getSelectRelationPod(value, objPods)
+	}
+	return objPods
+}
+
+//get an kubernetes resources's relation pods
+func (c *Client) getSelectRelationPod(info *resource.Info, objPods map[string][]api.Pod) (map[string][]api.Pod, error) {
+	if info == nil {
+		return objPods, nil
+	}
+
+	err := info.Get()
+	if err != nil {
+		return objPods, err
+	}
+
+	versioned, err := c.AsVersionedObject(info.Object)
+	if runtime.IsNotRegisteredError(err) {
+		return objPods, nil
+	}
+	if err != nil {
+		return objPods, err
+	}
+
+	selector, err := getSelectorFromObject(versioned)
+
+	c.Log("getSelectRelationPod selector: %+v", selector)
+	if err != nil {
+		return objPods, err
+	}
+	client, _ := c.ClientSet()
+
+	pods, err := client.Core().Pods(info.Namespace).List(metav1.ListOptions{
+		FieldSelector: fields.Everything().String(),
+		LabelSelector: labels.Set(selector).AsSelector().String(),
+	})
+	if err != nil {
+		return objPods, err
+	}
+
+	for _, pod := range pods.Items {
+		c.Log("get select relation pod: %s/%s", pod.Namespace, pod.Name)
+		if pod.APIVersion == "" {
```

Comment:
I don't think you need to check the API version and kind here. You should have a versioned object at this point. Did you run into an error with this while testing?

Reply:
When I was debugging, I found that when the API version is "v1", it return "". I don't know the deeper reason after I had done a lot of effort, so I gave it an default Version "v1" here.